fs-storage/src/utils.rs

```diff
@@ -51,6 +53,56 @@ where
     Ok(data)
 }
 
+/// Writes a serializable value to a file and returns the timestamp of the write
+pub fn write_json_file<T: Serialize>(
+    path: &Path,
+    value: &T,
+    time: SystemTime,
+) -> Result<()> {
+    let mut file = File::create(path)?;
+    file.write_all(serde_json::to_string_pretty(value)?.as_bytes())?;
+    file.flush()?;
+
+    file.set_modified(time)?;
+    file.sync_all()?;
+
+    Ok(())
+}
+
+pub fn extract_key_from_file_path<K>(
+    label: &str,
+    path: &Path,
+    include_extension: bool,
+) -> Result<K>
```

Comment:
Thanks for moving this function to the utils module. It definitely feels like the right place for it.

Could you also add a doc comment for it?